src/qr.jl

```diff
@@ -2,20 +2,203 @@ _thin_must_hold(thin) =
     thin || throw(ArgumentError("For the sake of type stability, `thin = true` must hold."))
 import Base.qr
 
-function qr(A::StaticMatrix, pivot::Type{Val{true}}; thin::Bool=true)
+
+"""
+    qr(A::StaticMatrix, pivot=Val{false}; thin=true) -> Q, R, [p]
+
+Compute the QR factorization of `A` such that `A = Q*R` or `A[:,p] = Q*R`, see [`qr`](@ref).
+This function is not support `thin=false` keyword option due to type inference instability.
+To use this option call `StaticArrays._qr(Size(A), A, pivot, Val{false})` instead.
```

Comment:
Typo: "This function is not" -> "This function does not".

I'd also suggest it's best to remove the recommendation to use `_qr` - it's not a public API, and I don't think we want to support it being stable going forward ;-) Instead, we can leave it as an unsupported hack, or add a new supported version of `qr` with a positional argument for the `thin=Val(false)` option.  If you don't need it, I suggest just removing the reference is fine for now.